javascript/js-declarations/src/org/jetbrains/dukat/js/declarations/JSModuleDeclaration.kt

```diff
@@ -0,0 +1,10 @@
+package org.jetbrains.dukat.js.declarations
+
+import org.jetbrains.dukat.js.declarations.toplevel.JSDeclaration
+
+data class JSModuleDeclaration(
+        val moduleName: String,
+        val fileName: String,
+        val exportDeclarations: MutableSet<JSDeclaration>,
+        val topLevelDeclarations: MutableMap<String, JSDeclaration>
```

Comment:
for consistency (with other targets) I suggest to call this just declarations.
Also, Map, generally speaking, does not guarantee us that the insertion order is preserved. 
It looks to me that we can have two options here:

1.  Switch to using special kind of Map that preserves the order
2. To use just lists and build whatever caching structures on demand. 

I'm very pro option 2 for a bunch of reasons, most important to me is the fact that we don't have this approach in other targets. And even if we want to move in that direction - let's move in a coordinated attempt. Later )

@trilis  what's are your thought on this?